compilerlibs/Makefile.compilerlibs

```diff
@@ -279,23 +279,24 @@ OPTTOPLEVEL_CMI=toplevel/topcommon.cmi toplevel/native/topeval.cmi \
 
 TOPLEVEL_SHARED_MLIS = topeval.mli trace.mli topmain.mli
 
-$(TOPLEVEL_SHARED_MLIS:%.mli=toplevel/byte/%.cmi):\
+$(addprefix toplevel/byte/, $(TOPLEVEL_SHARED_MLIS:%.mli=%.cmi)):\
 toplevel/byte/%.cmi: toplevel/%.cmi
 	cp $< toplevel/$*.mli $(@D)
 
-$(TOPLEVEL_SHARED_MLIS:%.mli=toplevel/native/%.cmi):\
+$(addprefix toplevel/native/, $(TOPLEVEL_SHARED_MLIS:%.mli=%.cmi)):\
 toplevel/native/%.cmi: toplevel/%.cmi
 	cp $< toplevel/$*.mli $(@D)
 
 beforedepend::
-	cp $(TOPLEVEL_SHARED_MLIS:%=toplevel/%) toplevel/byte
-	cp $(TOPLEVEL_SHARED_MLIS:%=toplevel/%) toplevel/native
+	cp $(addprefix toplevel/, $(TOPLEVEL_SHARED_MLIS)) toplevel/byte
+	cp $(addprefix toplevel/, $(TOPLEVEL_SHARED_MLIS)) toplevel/native
+
+TOPLEVEL_SHARED_HEADERS := \
+  $(TOPLEVEL_SHARED_MLIS) $(TOPLEVEL_SHARED_MLIS:%.mli=%.cmi)
 
 partialclean::
-	rm -f $(TOPLEVEL_SHARED_MLIS:%.mli=\
-	        toplevel/byte/%.mli toplevel/byte/%.cmi)
-	rm -f $(TOPLEVEL_SHARED_MLIS:%.mli=\
-	        toplevel/native/%.mli toplevel/native/%.cmi)
+	rm -f $(foreach ext, byte native, \
+    $(addprefix toplevel/$(ext)/, $(TOPLEVEL_SHARED_HEADERS)))
```

Comment:
Alternative proposal: define `TOPLEVEL_SHARED_CMIS` after `TOPLEVEL_SHARED_MLIS` (you can use it in the copy rules), and use `$(TOPLEVEL_SHARED_MLIS) $(TOPLEVEL_SHARED_CMIS)` instead of the (now-useless and not-so-well named anyway) `TOPLEVEL_SHARED_HEADERS` variable.

Reply:
It's a good idea, thanks - that's been changed. I also removed the somewhat gratuitous use of `$(foreach` (I think earlier on there were two loops...) and then figured that actually those targets are quite a bit clearer without all the `$(addprefix` too - what do you think?

Reply:
I think it's nice; merged.